drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -250,17 +257,14 @@ end
 local function find_default_endpoint(device, component)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
-  local all_eps = {}
 
-  for _,ep in ipairs(switch_eps) do
-    table.insert(all_eps, ep)
-  end
+  -- use first button endpoint as main endpoint if one exists
```

Comment:
I think it might be worth it to add just a little more context on why the first button endpoint would be used as the main endpoint. You've mentioned some of the reasoning in your confluence doc, I think it would just be helpful to have a bit more context why this is the case in the code.

Reply:
Added a comment with an explanation in the latest commit!